cmd/internal/find/primary/meta.go

```diff
@@ -82,3 +86,440 @@ var metaPrimary = Parser.add(&primary{
+entries that are going to expire within the next N minutes/hours/days/weeks
+(e.g. security policies, tagged EC2 instances, user credentials, etc.)
+
+STRING PREDICATE:
+Any input that doesn't begin with a "+", "-", "!", "(", or ")" is treated as
+a string predicate. String predicates return false for any non-string values.
+Otherwise, they return true if v == input, where v is the current meta value
+that's being compared.
+
+Below are some examples of string predicates:
+
+  foo
+      Returns true if v == foo 
+
+  bar
+      Returns true if v == bar
+
+And here's an example of a string predicate being used in conjunction with
+an object predicate:
+
+  .owner Mike
+      Returns true if m['owner'] == "Mike"
+
+NOTE: String predicates are a bit limited right now. We plan on adding
+regex/glob support for string predicates, and possibly a way for users to
+specify quoted string literals. The latter's useful when the input begins with
+one of the dis-allowed characters "+", "-", "!", "(", ")".
+
+NEGATION SEMANTICS:
+This section describes each of the aforementioned predicates' negation semantics.
+The information here can be used to reason about input like "! .foo +1".
+
+  OBJECT PREDICATE:  
+  "! .foo p" == ".foo ! p"
+
+  ARRAY PREDICATE:
+  "! [?] p" == "[*] ! p"
+  "! [*] p" == "[?] ! p"
+  "! [n] p" == "[n] ! p"
+
+  NULL PREDICATE:
+  "! -null" == "-exists"
+
+  EXISTS PREDICATE:
+  "! -exists" == "-null"
+
+  BOOLEAN PREDICATE:
+  "! -true"  == "-false"
+  "! -false" == "-true"
+
+  NUMERIC PREDICATE:
+  "! N"  == "+N" -o "-N"       (returns v != N)
+  "! +N" == "-N" -o "N"        (returns v <= N)
+  "! -N" == "+N" -o "N"        (returns v >= N)
+
+  Note that here, N represents a number. For example, N could be specified as
+  "1", "1k", or "{1}".
+
+  TIME PREDICATE:
+  "! 1h"    == "+1h" -o "-1h"          (returns "d" != 1h)
+  "! +1h"   == "-1h" -o "1h"           (returns "d" <= 1h)
+  "! -1h"   == "+1h" -o "1h"           (returns "d" >= 1h)
+
+  where "d" = "FindStartTime - v". Similarly for "future" queries,
+
+  "! {1h}"  == "+{1h}" -o "-{1h}"      (returns "d" != 1h)
+  "! +{1h}" == "-{1h}" -o "{1h}"       (returns "d" <= 1h)
+  "! -{1h}" == "+{1h}" -o "{1h}"       (returns "d" >= 1h)
+
+  where "d" = "v - FindStartTime". Note that the above semantics imply that
+  a negated time predicate will still return false if "d < 0" (i.e.
+  time-mismatches still return false)
+
+NOTE: The above semantics imply that the negated predicates still return false for
+mis-typed values. For example, "! .foo p" returns false if v is not an object;
+"! [?] p" returns false if v is not an array, etc.
+
+EXAMPLES:
+This section contains various, real-world examples of the meta primary's usage. If
+you have an interesting example that you'd like to include here, then please feel
+free to submit a PR!
+
+In these examples, let "m" be the value of the entry's 'meta' attribute.
```

Comment:
Open to some additional examples here. I tried to see if `meta` could query some useful stuff from pods, but Kube doesn't seem to be working on my machine.